lib/msf/core/db_manager/host.rb

```diff
@@ -185,74 +185,78 @@ def report_host(opts)
   ::ActiveRecord::Base.connection_pool.with_connection {
     wspace = Msf::Util::DBManager.process_opts_workspace(opts, framework)
 
-    ret = { }
-
-    if !addr.kind_of? ::Mdm::Host
-      addr = Msf::Util::Host.normalize_host(addr)
+    begin
+      retry_attempts ||= 0
+      if !addr.kind_of? ::Mdm::Host
+        addr = Msf::Util::Host.normalize_host(addr)
 
-      unless ipv46_validator(addr)
-        raise ::ArgumentError, "Invalid IP address in report_host(): #{addr}"
-      end
+        unless ipv46_validator(addr)
+          raise ::ArgumentError, "Invalid IP address in report_host(): #{addr}"
+        end
 
-      if opts[:comm] and opts[:comm].length > 0
-        host = wspace.hosts.where(address: addr, comm: opts[:comm]).first_or_initialize
+        conditions = {address: addr}
+        conditions[:comm] = opts[:comm] if !opts[:comm].nil? && opts[:comm].length > 0
```

Comment:
What is `opts[:comm]` supposed to be? The old code only ran the `hosts` lookup if `opts[:comm]` was present, now it always runs. Is that ok?